Fetch/NoDataResponse.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  NoDataResponse.swift
+//  Fetch
+//
+//  Created by David Haynes on 14/12/2017.
+//  Copyright © 2017 David Hardiman. All rights reserved.
+//
+
+import Foundation
+
+/// Represents a response that has no data to parse
+public struct NoDataResponse {
+    enum NoDataResponseError: Error {
```

Comment:
Probably worth this being public otherwise it'd be hard to interrogate it? Also worth the status code being sent as a payload so you can check between errors you can do something about (400 might indicate a programming error, 500 temporary server side)?

Reply:
Good idea, will do that.